homeassistant/components/smappee/sensor.py

```diff
@@ -291,17 +313,36 @@ def device_class(self):
         """Return the class of this device, from component DEVICE_CLASSES."""
         return self._device_class
 
+    @property
+    def state_class(self):
+        """Return the state class of this device."""
+        return STATE_CLASS_MEASUREMENT
+
     @property
     def unit_of_measurement(self):
         """Return the unit of measurement of this entity, if any."""
         return self._unit_of_measurement
 
+    @property
+    def last_reset(self):
+        """Return the time when the sensor was last reset, if any."""
+        if self._sensor in ("power_today", "solar_today", "alwayson_today", "switch"):
+            return start_of_local_day()
+        elif self._sensor in ("power_current_hour", "solar_current_hour"):
+            datetime_now = utcnow()
```

Comment:
For current hour, you should start your calculation with `now()`, as it represents the current time zone. We shouldn't assume that a timezone is always exact number of whole hours different from UTC. It could be 4.5 etc. You can use the `as_utc` helper to conver it back to UTC afterwards.